nvme_gw_server.py

```diff
@@ -167,8 +167,6 @@ def start_spdk(self):
 
         try:
             self.spdk_process = subprocess.Popen(cmd,
-                                                 stderr=subprocess.PIPE,
-                                                 stdout=subprocess.PIPE,
```

Comment:
Doesn't SPDK also log to syslog?  Looking at https://spdk.io/doc/log_8h.html, there is a "print" (stderr) log level and a "syslog" log level and `JSONRPCClient` that is used by the gateway takes `log_level` parameter as well.  How do these three interact?

If by default all messages go to syslog and only some messages (e.g. `SPDK_LOG_ERROR`) go to stderr it might make more sense to redirect to `DEVNULL` and rely on syslog until the proper logging infrastructure is in place.

Just a thought -- not redirecting (i.e. inheriting file handles from the gateway) as proposed is fine.

Reply:
Just checked, yes it logged both to syslog and stderr. It seems there are more debug messages in stderr though, I have not investigated this.

Anyway, taking that the gateway (nvme_gw_server.py) currently logs to stderr, I would prefer the nvmf_tgt would do the same. It looks like it is easier to read the logs this way. And this is also useful when running in a k8s container.

Reply:
OK, makes sense to me.